### ISSUE TITLE
Game mechanics idea collection

### ISSUE THREAD

**Issue by Sesu8642** [Sesu8642/FeudalTactics#4]
Here are some ideas for game mechanics that I had or received as feedback from others. Generally, I want to preserve the Slay mechanics as is in some way. New mechanics could be part of a separate mode or even a successor to the game.
- new unit: catapults that can destroy units from the distance
- trees being useful (e.g. by giving wood)
- destroying a capital gives money
- capitals get stronger when they have a certain amount of money
- limited movement range for units

**Comment by jmizv** [comment]
Hi @Sesu8642 
really cool game, thanks.
I would suggest that one has some log messages or an abstract of what happened since the last round. I mean sometime I'm puzzled why my units are gone. A little message like "Some of your units were lost due to insufficient money" would be helpful.

**Comment by Sesu8642** [comment]
Hi @jmizv,
thanks for your feedback! I will try to visualize this a bit better in the next update. I don't want to annoy the player with too many dialogs or too much text. Some of the ideas I have:
- when you can't pay your units:
  - finance text becomes red and maybe contains a warning next to it
  - units hold signs with sad smiley faces instead of weapons
- when a unit dies, a gravestone appears (like in Slay)

If you have other/better ideas, please let me know.

**Comment by bioderm** [comment]
Hello @Sesu8642,
first of all thank you very much for this wonderful game.

Here are some suggestions from my side:
- Trees block certain units (e.g. knight and baron).
- Capitals integrate into the area from where the unit is moved (currently it is the other way round).
- Units can be dismissed
- the capital becomes stronger due to the size of the area
- randomly placed gold to collect (especially on the territory of weak players) 

I wish you continued success with the game/project.

**Comment by Sesu8642** [comment]
Hello @bioderm,
thanks for your ideas! Would you mind elaborating the first two?
- What does "block" mean? Can existing units no longer be moved when there is a tree?
- Like conquering the capital?

**Comment by aligator** [comment]
I know an even older version on the Atari. The rules were basically the same with two major differences: (if I remember correctly)
* when someone died a gravestone was created and in the next round the gravestone was always converted to a tree.
* the base houses in each kingdom did not protect the neighbour fields. 

You can get the demo here https://www.janatari.de/atari-firebee/firebee-games/slay-1-0/
And play it with an atari tt / falcon emulator (like aranym)
Sadly there no product key available anymore. Even the original Dev doesn't have it anymore... (https://forum.atari-home.de/index.php?topic=9865.0 )

**Comment by Sesu8642** [comment]
@aligator Interesting! I wasn't aware of this version at all. Are you sure it is older? It seems like it released in 99 (see https://www.maedicke.de/atari/download.htm) while O'Connor's version released in 95.

I actually don't own O'Connor's Slay either. I looked at some YouTube videos to see how the mechanics work. I probably missed the fact that the gravestones resulted in trees. A colleague told me about this as well. According to him, I missed another detail: The spreading behavior of regular trees and palms differs. But I'd have to analyze how it works exactly.

**Comment by aligator** [comment]
No I am not sure who created Slay first ^^ I thought it was older, but you may be right, that actually O'Connor was first. I bought his some time ago, but after I already knew the Atari (demo) version. Also the atari version seems to be "Rel. 2" according to your link, so there may have been another version :-)

However I was very happy to find your version in F-Droid because I played the atari demo when I was a child.

I think the gravestone-trees are somewhat important, as you have to look out for that.
e.g. 
Your version: you have a tier2 soldier on a 2-field kingdom -> it dies -> some rounds later you may get enough money to respawn a new one.  
With grave-trees: you have a tier2 soldier on a 2-field kingdom -> it dies -> you get trees -> you won't get money to respawn a new one.
That may change the gameplay in some situations 

> The spreading behavior of regular trees and palms differs
Interesting, that's something I wasn't aware of^^

**Comment by bioderm** [comment]
> 
>     * What does "block" mean? Can existing units no longer be moved when there is a tree?
>
These units cannot be placed on a field with a tree. Thus, even the weak units remain important later.
> 
>     * Like conquering the capital?
>
No. I am referring to the situation when you unite two kingdoms. As an example, let's take A and B. If you join B from A, the capital will be moved to B. This should remain the other way around. Often one moves from the large well-secured territory to a weak one.

**Comment by Sesu8642** [comment]
@bioderm I see. Thanks for clarifying.

**Comment by Sesu8642** [comment]
Improvement idea i received via email:
If there is not enough money to pay all units, not all of them should die. Instead, the player should be able to select which ones to let off.

**Comment by d-albrecht** [comment]
On the one hand, I can see that not being able to pay the units is a corner case where this is realistic, but on the other hand it clashes with the general rule that no units can be laid off willingly.

However I could see an argument for only letting some units die up to the point where the remaining can be paid. But this again, should work without user input. Because (a) additional options would just slow down the game and (b) this would again contradict the "no manual lay off" rule. But it would have the advantage that you had some units left. As this would have the most effect, I would suppose that the game tries to kill ALL barons . If this works, then you could keep everything else. otherwise all knights would get killed. Then - if still necessary - all spearmen and finally all peasants.

I said ALL units of the same strength because this would reduce randomness. And it would still be somewhat realistic. Imagine your kingdom is bankrupt, why should some baron decide to stay (without guaranteed payment) and others would leave. But as the units only die once your turn starts and you are able to immediate move them around between some units dying and your opponents being able to attack you, laying off some units could work as well.

The problem with this general idea of keeping some units is that it must be clearly communicated that this happened. Otherwise the players will be completely caught be surprise if some units have vanished while others remained.

**Comment by d-albrecht** [comment]
Another item for this idea collection:
Have a viewable history of a game. When it is my turn, I can enter some "History" mode in which I can step through the whole previous states step by step. Essentially showing me any AI step that I saw animated between turns. Such that I can check what happened without any time pressure. (And can debug AI behavior.)

**Comment by Sesu8642** [comment]
Idea by @YurishoSan, see #46:
"In cases where i can see I wont be able to sustain the whole army tbe next turn, it should be a valid tactic to decrese army size in advance as to not forfiet the entire army.

This will allow "rush" tactics that eat the money reserves for a direct attack, for example to target an enemy castle, and then cut down the army size before it implodes.

Another use for this mechanic is to decrese army size in order to facilitate a baron. When I know I need a baron i currently need to avoid making speerman, as they block me financialy from making the baron. killing the speerman when the time comes will allow me to make the baron when needed without secrificing my military power before hand. This at the cost of killing the spearman.

This last mechanic should be considered in opossition to the possibility of combining spearman to create a baron, in which case the cost of the spearmen will not be waisted."

**Comment by TheShadowOfHassen** [comment]
This isn't a game mechanic idea, but could you set it up theming? The slay I've played allows you to change the looks of the people tiles and buttons. I'd love to be able to change the art to penguins or pirates or other different styles of art

**Comment by Sesu8642** [comment]
@TheShadowOfHassen i like it but it's not a priority for me for now. It could work well with a "campaign". In the campaign the enemy uses the skin and when you beat the campaign, you get it as well as an option.

**Comment by TheShadowOfHassen** [comment]
Yeah, I understand, it was just an idea. The campaign integration is a cool idea though.

**Comment by Sesu8642** [comment]
Via email I received another vote for the ability to combine two spearmen to get a baron.

**Comment by Blop32** [comment]
Hi,
first off all thanks for the great game!
I really like the mechanics as it is. Not too complex in terms of variety of units, rules etc. It develops its complexity out of simplicity. Similar like go or checkers.
Perfect for a quick brain challenge on the go :-)
In particular the trees can be a real nuisance (which is good), compared to other variants of the game.

However if there'd be one vote I could give for improving the game it would be an improved algorithm to create the levels. It seems random and once you get the hang of the game there are basically loads of seeds that are pretty pointless (in very hard mode). Either its a sure win or an inevitable loose. There are seeds where you can do whatever you want, it will lead to an overwhelmingly powerful enemy or vice versa.
Right now I end up browsing through several seeds and trying to evaluate if its an interesting one.
These could be some ideas for improvement:
- providing each party the same number of tiles and the same amount of units at the beginning (currently it seems number of tiles are evenly allocated but randomly placed. Those parties with several tiles in one block get an advantage in terms of number of units for the first round)
- somehow make sure each party gets the same advantages in terms of starting position. No idea how that could look like in detail. But one almost always wins when starting in a corner with a sufficiently large patch and one almost always looses when placed in the center (multiple front war...). Maybe provide encircled patches some extra bonus (additional tiles or units)
- The loose map density is pretty pointless with the current mechanics. Its basically a point an click scheme where one progresses linearly towards predictable victory or defeat. With ranged units or boats that could change the situation (but make the game overall more complicated. IMHO complexity emerging from simplicity is one of the main characteristics of the game that should be maintained).
Cheers!!

**Comment by Sesu8642** [comment]
@Blop32 thanks for your feedback! I moved your input to #35 because I think it fits there better.

**Comment by HasbersHz** [comment]
I don't know should i ask this here, but i don't want to create new issue. Is there will be big updates and when? I like the game, but i'm feeled bored, because i can tell from first look if i win here or not. Also i would like if "Loose"  be more loose and if you can change width of paths.

**Comment by Sesu8642** [comment]
@HasbersHz Since this is a hobby project, my time to spend on it is fairly limited. Same goes for the contributors. But as you can see, there are many open issues I would like to implement eventually. Improved map generation is one of them. Just don't expect any large "2.0" update. It will be more incremental.

**Comment by Sesu8642** [comment]
@NicolasToussaint I agree that the game rules shouldn't be more complicated than they are. If I decide to do major changes, I will do so in a way that at least preserves the original rule set as an option :)
Also: Thanks for your opinion on the other ideas!

**Comment by HasbersHz** [comment]
> @HasbersHz Since this is a hobby project, my time to spend on it is fairly limited. Same goes for the contributors. But as you can see, there are many open issues I would like to implement eventually. Improved map generation is one of them. Just don't expect any large "2.0" update. It will be more incremental.

Ok. I would like to implement some features, but i have many projects.

**Comment by Sesu8642** [comment]
@HasbersHz No worries. Should you find the time, just let me know :)

**Comment by HasbersHz** [comment]
Actually issue, why there is no community? I think spending nights playing this game damaged my brain, because i even making a memes about it.

**Comment by HasbersHz** [comment]
And also now I think that "very hard" is baby, because there is too many stupid decisions he is making. Like this two that i think many people encountered: "No baron", for some reason, even if it become a stalemate it is a logic decision; "Sawing the budget to death": "AI" spend budget of whole world for 20 barons, and waiting ten moves for their agonizing death.

**Comment by Sesu8642** [comment]
> I think spending nights playing this game damaged my brain

Sorry for that :smile: 
Where are you posting the memes? :D
I had the idea to create a community group somewhere too. Probably Matrix. Only issue is that very few people will know that it exists unless I put the link in the game. Which I will probably do but only with the next feature update and who knows when that will be ready :x

Very hard feeling like a baby is probably the result of you playing too much ;)
Yes, there are improvements to be made but there is a separate issue for that. #20

**Comment by KjmT6459** [comment]
Just played a game and tried surrounding an enemy stronghold to capture and keep it, unfortunately it disappeared.

I think it would be cool if on being surrounded, the building would convert to your side. 
I also like the idea of it just staying in place, like inland trees. I just like the look of extra stuff on the map, so I actually force trees often and strongholds would look neat too.

**Comment by Asmageddon** [comment]
As someone who loves thinking about game design and has been enjoying this game - I'm shocked at how much variety such simple rules create - lemme offer a few of my thoughts and ideas.

## Thoughts

> destroying a capital gives money

I do not think this is a good idea. This game is all about economic advantage, and destroying a capital already puts you ahead by a lot and starvation mechanics compound it even more.

On the contrary, I think the game should be more graceful with starvation mechanics, either A) pay units one-by-one(starting from smallest), only starving the ones that cannot be paid, not all, B) make the units start exhausted(maybe highlighted in red?) when starving, and only die the next turn, and possibly even C) both.

In fact, I think C would be the best option, since B would still be unrecoverable as you can't command even peasants, and unlike A, C doesn't free up your income immediately, in spirit with the "you have to commit to upgraded units" philosophy.

> capitals get stronger when they have a certain amount of money

What about just letting players upgrade them by building a Castle on top of them? 

> limited movement range for units

I think this is an extremely solid idea, since right now your kingdom is threatened even if opponent's units are 20 tiles away from your borders, which creates a pretty strange dynamic sometimes. Having to commit units to a given front would IMO improve the game a lot.

> Trees block certain units (e.g. knight and baron).

I like this. Make them unable to pass through or capture trees.

> Units can be dismissed

Maybe worth experimenting with, but having to commit to upgrades adds weight to the decision, though OTOH if you need higher-tier units, enemies probably also have them, and you still pay the cost of the unit. I'd still be worried it'd disadvantage weaker players.

UI-wise, have an icon for "Dismiss" that will work on whatever can be deconstructed, be it units or buildings.

> randomly placed gold to collect (especially on the territory of weak players)

I like this. For a variant, what if instead of gold, it was wild crops(wheat,pumpkins,whatever) that can be upgraded to an income-boosting farm by a peasant?

## Ideas

First off, I believe that the minimalism of this game is very good, and should be preserved even if you add more buildings and units.

One interesting way to do it that I see would be to have a pool of possible units,buildings, etc. but only have a few additional ones enabled at any given time- by default, randomly, and with the same options available to the same player.

### Game rules

What if merging kingdoms didn't destroy one of the capitals, but only downgraded it to a village - which preferentially becomes a new capital if the kingdoms are split again, and which can be captured.

I think it could add some extra depth if you also made cities generate extra income(1 for village, 2 for capital?) and made it possible to seize enemy villages.

Another potentially interesting extension to the game would be to have rivers/chasms/walls between some cells that can only be passed by units that start their turn next to it, to create some soft natural borders.

### Buildings

**Economy buildings**:
- Bank - 20g, generates 5/3/2 for the 1st/2nd/3rd+ you own(build order) so you can't just snowball your economy. 
- Farm(ver A) - 12g, creates fields around itself, generates 1g per two fields. Two farms can't work the same field. Captureable.
- Farm(ver B) - 8g, generates 1g, you can't build two next to each other. Captureable.
- Lumber Mill(ver A) - costs 8 gold, generates 1+N gold for N>1 trees.
- Lumber Mill(ver B) - costs 12g, generates 2g for each surrounding tree. Not sure which is better. 

**Military buildings**:
- Wall(ver B) - 6g, 2 shields only on itself, your own units can't pass through it either.
- Underground Fortress - 25g. Like a fortress, but not destroyed when captured, you can get it back.
- Fort - 10g. Generates 1 shield on its own, 2 with unit on top of it but also halves its upkeep.

**Other buildings**:
- Road - 4g. Units can travel twice as far on it. Captureable.

**Units**:
- Archer/Crossbowman(T2/T3) - can kill units or neutralize castles from a distance without moving onto them.
- Rider - can travel twice as far as other units.
- Siege Engine - low movement, cannot capture on its own but lowers shield values of nearby enemy tiles(except units themselves). E.g. lets a Peasant capture tiles next to another Peasant, or capture Castles with a Spearman.
- Infiltrator - T1. Slightly lower movement than other units, can move through 1 or 2 tiles of enemy territories and doesn't die the first turn it can't be paid, so it can e.g. reach your other kingdoms, cut off supply, etc. Cannot move through units.

My concerns would be that most economy buildings would advantage players with larger, more defensible territories more, and that they might potentially turn into compounding advantage.

**Comment by Sesu8642** [comment]
@Asmageddon thanks for your many ideas! Most of them I'd consider mainly for a potential Feudal Tactics 2 as they are quite significant changes.

That said, one of the next major features is going to be a campaign of hand-crafted levels. Expanding on that, there could be additional campaigns at some point that each evolve around one or two new mechanics.
